ert3/config/_parameters_config.py

```diff
@@ -105,27 +105,53 @@ def __len__(self) -> int:
         return len(self.__root__)
 
 
+def _ensure_valid_size(size: int) -> Optional[int]:
+    if size is not None and size <= 0:
+        raise ValueError("Size cannot be <= 0")
+
+    return size
+
+
 class _ParameterConfig(_ParametersConfig):
     name: str
     type: Literal["stochastic"]
     distribution: Union[_GaussianDistribution, _UniformDistribution]
-    variables: _VariablesConfig
+    variables: Optional[_VariablesConfig] = None
+    size: Optional[int] = None
+
+    @root_validator
+    def _ensure_variables_or_size(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if (values.get("variables") is None and values.get("size") is not None) or (
+            values.get("variables") is not None and values.get("size") is None
+        ):
+            return values
+
+        raise AssertionError("Parameter group cannot contain both variables and size")
 
     @validator("name")
     def _ensure_valid_group_name(cls, value: Any) -> str:
         return _ensure_valid_name(value)
 
+    @validator("size")
+    def _ensure_valid_group_size(cls, value: Any) -> Optional[int]:
+        return _ensure_valid_size(value)
+
     def as_distribution(self) -> ert3.stats.Distribution:
         dist_config = self.distribution
-        index: ert.data.RecordIndex = tuple(self.variables)
+        assert (self.variables is None and self.size is not None) or (
```

Comment:
xor could be used here as well

Reply:
The assert was removed.

Reply:
Adapted range to the `x_uncertainties`. Default for no `x_uncertainties` is still `range(10)`.